cmd/local-gadget/trace/sni.go

```diff
@@ -0,0 +1,133 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	commontrace "github.com/inspektor-gadget/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/networktracer"
+	sniTracer "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/sni/tracer"
+	sniTypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/sni/types"
+	localgadgetmanager "github.com/inspektor-gadget/inspektor-gadget/pkg/local-gadget-manager"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+func newSNICmd() *cobra.Command {
+	var commonFlags utils.CommonFlags
+
+	// The SNI gadget works in a different way than most gadgets: It
+	// attaches a new eBPF program to each container when it's
+	// created instead of using an eBPF map with the mount
+	// namespaces IDs to filter the events. For this reason we can't
+	// use the TraceGadget implementation here.
+	runCmd := func(*cobra.Command, []string) error {
+		localGadgetManager, err := localgadgetmanager.NewManager(commonFlags.RuntimeConfigs)
+		if err != nil {
+			return commonutils.WrapInErrManagerInit(err)
+		}
+		defer localGadgetManager.Close()
+
+		// local-gadget is designed to trace containers, hence enable this column
+		cols := sniTypes.GetColumns()
+		col, _ := cols.GetColumn("container")
+		col.Visible = true
+
+		parser, err := commonutils.NewGadgetParserWithRuntimeInfo(&commonFlags.OutputConfig, cols)
+		if err != nil {
+			return commonutils.WrapInErrParserCreate(err)
+		}
+
+		eventCallback := func(container *containercollection.Container, event sniTypes.Event) {
+			baseEvent := event.GetBaseEvent()
+			if baseEvent.Type != eventtypes.NORMAL {
+				commonutils.HandleSpecialEvent(baseEvent, commonFlags.Verbose)
+				return
+			}
+
+			// Enrich with data from container
+			if !container.HostNetwork {
+				event.Namespace = container.Namespace
+				event.Pod = container.Podname
+				event.Container = container.Name
+			}
+
+			switch commonFlags.OutputMode {
+			case commonutils.OutputModeJSON:
+				b, err := json.Marshal(event)
+				if err != nil {
+					fmt.Fprint(os.Stderr, fmt.Sprint(commonutils.WrapInErrMarshalOutput(err)))
+					return
+				}
+
+				fmt.Println(string(b))
+			case commonutils.OutputModeColumns:
+				fallthrough
+			case commonutils.OutputModeCustomColumns:
+				fmt.Println(parser.TransformIntoColumns(&event))
+			default:
+				fmt.Fprint(os.Stderr, commonutils.WrapInErrOutputModeNotSupported(commonFlags.OutputMode))
```

Comment:
We don't need to check this case anymore -> https://github.com/inspektor-gadget/inspektor-gadget/pull/1135